library/aesni.c

```diff
@@ -52,8 +52,8 @@ int mbedtls_aesni_has_support(unsigned int what)
 
     if (!done) {
 #if MBEDTLS_AESNI_HAVE_CODE == 2
-        static unsigned info[4] = { 0, 0, 0, 0 };
-#if defined(_MSC_VER)
+        static int info[4] = { 0, 0, 0, 0 };
+#if defined(_WIN32)
```

Comment:
That's difference between GCC and MSC. I prefer change ln 36 to `#if !defined(_MSC_VER)`

Reply:
This might not work when using clang compiler driver under Visual Studio, which is becoming more common.

(When clang is run under Linux, it sets `__GNUC__`, but when run as compiler driver under Visual Studio, it sets `_MSC_VER`. This definitely causes a problem for inline assembly (since clang format is not the same as MSVC format) - I need to test the intrinsics)

Reply:
@tom-cosgrove-arm , Yes, that might be a problem.  Does that mean both macro will be defined in this case ?

Reply:
> This might not work when using clang compiler driver under Visual Studio, which is becoming more common.

I've tested it, and no related issue. When clang is running in MSVC compatibility mode, it sets `_MSC_VER` and provide a compatible `intrin.h`.

Reply:
Thanks for checking